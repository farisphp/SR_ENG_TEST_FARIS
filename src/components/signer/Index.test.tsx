import React from 'react';
import { render, cleanup, screen } from '@testing-library/react';
import SignerIndex from './Index';
import { BrowserRouter, Switch, Route} from 'react-router-dom'; 

afterEach(cleanup)

describe('Props test', () => {
  test('Should display Penandatanganan SPT', () => {
    render(
        <BrowserRouter>
            <Switch>
                <Route path={'/'} exact component={SignerIndex} />
            </Switch>
        </BrowserRouter>
    );
    const elements = screen.getAllByText("Penandatangan SPT");

    elements.forEach((element) => {
        expect(element.textContent).toBe("Penandatangan SPT");
    });
  })
})

describe('Snapshot test', () => {
    test('Renders correctly.', () => {
        render(
            <BrowserRouter>
                <Switch>
                    <Route path={'/'} exact component={SignerIndex} />
                </Switch>
            </BrowserRouter>
        );
        const elements = screen.getAllByText("Penandatangan SPT");
    
        expect(elements).toMatchSnapshot()
    })
})