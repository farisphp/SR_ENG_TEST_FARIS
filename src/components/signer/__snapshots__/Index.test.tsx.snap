// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshot test Renders correctly. 1`] = `
Array [
  <h2
    class="header"
  >
    Penandatangan SPT
  </h2>,
]
`;
